src/gsi/cplr_get_fv3_regional_ensperts.f90

```diff
@@ -393,40 +409,95 @@ subroutine get_fv3_regional_ensperts_run(this,en_perts,nelen,ps_bar)
-                 else if( if_model_dbz )then
-                    call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz,   &
-                                                        g_ql=ql,g_qi=qi,g_qr=qr,g_qs=qs,g_qg=qg,g_qnr=qnr,g_w=w,g_dbz=dbz)
-                 end if
-              end if
+          
+              ! There are three options to control the list of variables that
+              ! will be read in along with the basic variables, ps,u,v,tv,rh,oz.
+              ! Here the 6 cases that are considered in
+              ! the current applications are listed as of Oct 20 2023.
+
+              ! default: all the three options ( l_use_dbz_directDA, if_model_dbz, if_model_fed) are turned off .i.e.,
+              !          if(.not. (l_use_dbz_directDA .or. if_model_dbz .or.  if_model_fed ))
+              i_caseflag=0  ! read in ps,u,v,tv,rh,oz
+
+              ! only l_use_dbz_directDA is true
+              if (l_use_dbz_directDA     .and. .not.if_model_dbz .and. .not.if_model_fed) i_caseflag=1
+
+              ! only if_model_dbz is true
+              if(.not.l_use_dbz_directDA .and.      if_model_dbz .and. .not.if_model_fed) i_caseflag=2
+
+              ! only if_model_fed is true
+              if(.not.l_use_dbz_directDA .and. .not.if_model_dbz .and. .not.if_model_fed) i_caseflag=3
+
+              ! l_use_dbz_directDA=.true. and if_model_fed=.true.
+              if(l_use_dbz_directDA      .and. .not.if_model_dbz .and.      if_model_fed) i_caseflag=4
+
+              ! if_model_dbz=.true. and if_model_fed=.true.
+              if(.not. l_use_dbz_directDA.and.      if_model_dbz .and.      if_model_fed) i_caseflag=5
+
+              select case (i_caseflag)
+                case (0)
+                  call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz)
+                case (1)
+                  call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz,   &
+                            g_ql=ql,g_qi=qi,g_qr=qr,g_qs=qs,g_qg=qg,g_qnr=qnr,g_w=w)
+                case (2)
+                  call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz,   &
+                            g_ql=ql,g_qi=qi,g_qr=qr,g_qs=qs,g_qg=qg,g_qnr=qnr,g_w=w,g_dbz=dbz)
+                case (3)
+                  call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz,   &
+                            g_ql=ql,g_qi=qi,g_qr=qr,g_qs=qs,g_qg=qg,g_w=w,g_fed=fed)
+                case (4)
+                  call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz,   &
+                            g_ql=ql,g_qi=qi,g_qr=qr,g_qs=qs,g_qg=qg,g_qnr=qnr,g_w=w,g_fed=fed)
+                case (5)
+                  call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz,   &
+                            g_ql=ql,g_qi=qi,g_qr=qr,g_qs=qs,g_qg=qg,g_qnr=qnr,g_w=w,g_dbz=dbz,g_fed=fed)
+              end select
            end if
- 
+
            if( parallelization_over_ensmembers )then 
```

Comment:
@wangym1111 
Do you have an idea why this section [if( parallelization_over_ensmembers )then ] doesn't use l_use_dbz_directDA? 
@TingLei-NOAA FYI. 

Thanks, 
Hongli

Reply:
Hi @hongli-wang , I guess I can answer the question. 

To the best of my recollection, we did not test the 'l_use_dbz_directDA' with parallel 'read' capabilities in GSI during our code implementation. This explains its current configuration... Thanks,

Reply:
@junpark217 
In this case, if l_use_dbz_directDA is true, it shouldn't use the read_over_ens (true). 
Thanks for the reply! 

Hongli

Reply:
> @wangym1111 Do you have an idea why this section [if( parallelization_over_ensmembers )then ] doesn't use l_use_dbz_directDA? @TingLei-NOAA FYI.
> 
> Thanks, Hongli

@hongli-wang Sorry for the late response. I missed the message. I think @junpark217 explained it well.

Reply:
@wangym1111 
Thanks for your confirmation. Thanks again for your prevous comments/suggestions that improved this PR. Please let me know if you have other comments. 

Best, 
Hongli